src/middlewares.js

```diff
@@ -144,6 +146,33 @@ function handleParseHeaders(req, res, next) {
     });
 }
 
+function httpAuth(req) {
+  var appId, masterKey, javascriptKey;
+  var credentialsRegExp = /^ *(?:[Bb][Aa][Ss][Ii][Cc]) +([A-Za-z0-9\-\._~\+\/]+=*) *$/
+  var appIdKeyRegExp = /^([^ :]*):([Jj][Aa][Vv][Aa][Ss][Cc][Rr][Ii][Pp][Tt][-][Kk][Ee][Yy]=)?(.*)/
+ // get header
+  var header = (req.req || req).headers.authorization
+  // parse header
+  var match = credentialsRegExp.exec(header || '')
```

Comment:
may be just check that `indexOf('Basic: ') == 0` instead of the convoluted regexp